modules/doi/includes/utilities.inc

```diff
@@ -864,3 +862,51 @@ function islandora_doi_get_mods($id) {
 
   return islandora_doi_translate($source_string, $content_type);
 }
+
+/**
+ * Filter HTML tags from imported XML metadata.
+ *
+ * @param DOMNode $node
+ *   The node containing metadata that should be converted into a string,
+ *   possibly containing markup.
+ *
+ * @return string 
+ *   The node's inner content with markup escaped or stripped.
+ */
+function islandora_doi_filter_html($node, $action = NULL) {
+  if (!$node) {
+    return "";
+  }
+  else {
+    $text  = '';
+    foreach ($node->childNodes as $child) {
+      $text .= $child->ownerDocument->saveXML($child);
+    }
+
+    $text = preg_replace('/\n\s+/', ' ', $text);
+
+    if (!$action) {
+      $action = variable_get('islandora_doi_handle_imported_metadata', 'strip');
+    }
+
+    switch ($action) {
+      case 'strip':
+        $text = strip_tags($text);
+        break;
+
+      case 'encode':
+        $text = htmlspecialchars($text);
+        break;
+
+      default:
+        drupal_set_message(t('@var is not a recognized option for handling HTML in imported DOI metadata.', array('@var' => $action)), 'error');
+        break;
+    }
+
+    // Double encode ampersands regardless of prior filtering
+    $text = str_replace(' & ', ' &amp; ', $text);
+    $text = str_replace(' &amp; ', ' &amp;amp; ', $text);
```

Comment:
Do we really need these three lines? A quick [test](http://localhost:8000/devel/php),
```php
$text = "<i>Needles’ &amp; Pins&#x2019;</i>";
dpm(print_r(strip_tags($text), TRUE)); // Needles’ &amp; Pins&#x2019;
dpm(print_r(htmlspecialchars($text), TRUE)); // &lt;i&gt;Needles’ &amp;amp; Pins&amp;#x2019;&lt;/i&gt;
```
seems to indicate that we obtain already what we actually want:
- in _strip_ mode: something that will translate to un-encoded plain text on display
- in _encode_ mode: encoded HTML that will translate to HTML source (use [`htmlspecialchars_decode()`](https://secure.php.net/manual/en/function.htmlspecialchars-decode.php) to view - actually we might want to give this hint to users)

Note also from TravisCI: The comment needs to end with a full stop.